deltachat-ios/Chat/Views/WebxdcPreview.swift

```diff
@@ -0,0 +1,85 @@
+import UIKit
+import DcCore
+
+public class WebxdcPreview: UIView {
+    
+    lazy var imagePreview: UIImageView = {
+        let view = UIImageView()
+        view.contentMode = .left
+        view.translatesAutoresizingMaskIntoConstraints = false
+        view.layer.cornerRadius = 8
+        view.clipsToBounds = true
+        return view
+    }()
+    
+    lazy var titleView: UILabel = {
+        let view = UILabel()
+        view.font = UIFont.preferredBoldFont(for: .body)
+        view.translatesAutoresizingMaskIntoConstraints = false
+        view.numberOfLines = 1
+        view.lineBreakMode = .byTruncatingTail
+        isAccessibilityElement = false
+        return view
+    }()
+    
+    lazy var subtitleView: UILabel = {
+        let view = UILabel()
+        view.font = UIFont.preferredFont(forTextStyle: .body)
+        view.translatesAutoresizingMaskIntoConstraints = false
+        view.numberOfLines = 3
+        isAccessibilityElement = false
+        return view
+    }()
+    
+    lazy var containerStackView: UIStackView = {
+        let view = UIStackView(arrangedSubviews: [imagePreview, titleView, subtitleView])
+        view.axis = .vertical
+        view.alignment = .leading
+        view.spacing = 6
+        view.translatesAutoresizingMaskIntoConstraints = false
+        return view
+    }()
+    
+    convenience init() {
+        self.init(frame: .zero)
+
+    }
+    
+    required init?(coder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+
+    public override init(frame: CGRect) {
+        super.init(frame: frame)
+        self.setupSubviews()
+    }
+    
+    private func setupSubviews() {
+        addSubview(containerStackView)
+        containerStackView.fillSuperview()
+    }
+    
+    public func configure(message: DcMsg) {
+        imagePreview.image = message.getWebxdcIcon()?.sd_resizedImage(with: CGSize(width: 175, height: 175), scaleMode: .aspectFill)
+        titleView.text = message.getWebxdcName()
+        subtitleView.text = message.getWebxdcSummary()
```

Comment:
this is pretty inefficient as each line results in db-access and unzipping.

i'd just call `info = message.getWebxdcInfoDict()` once and check for `info["name"]`, `info["icon"]`  etc. then.

at least `getWebxdcIcon()` and `getWebxdcSummary()` can be deleted then, probably also`getWebxdcName()` if we add and additional line to  `WebxdcViewController.viewDidLoad()`.

that would be 3x faster and less code already now - and in the future, there may be even more things to check. having an explicit getter seem to be too much overhead unless the thing is really used often, and not only one or two times.

Reply:
the efficiency gain is probably neglectable, plus the code will  be less readable. I revert the getters anyways.